akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -311,7 +311,16 @@ private[stream] class VirtualSubscriber[T](val owner: VirtualPublisher[T]) exten
  */
 private[stream] class VirtualPublisher[T]() extends Publisher[T] {
   @volatile var realPublisher: Publisher[T] = null
-  override def subscribe(s: Subscriber[_ >: T]): Unit = realPublisher.subscribe(s)
+  override def subscribe(s: Subscriber[_ >: T]): Unit = {
+    val sub = realPublisher.subscribe(s)
+    realPublisher match {
+      case _: SupportingMultipleSubscribers   ⇒
```

Comment:
Isn't this completely Akka specific now? I mean, if you take a Source(thirdPartyPublisher).runWith(Sink.publisher) then this will remove the multi-subscribe support from thirdPartyPublisher.